roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
When I first saw your PR a while back I placed my hope in being able to test the `ansible_connection` variable, but in my tests it never seemed defined unless the hosts file (inventory file) explicitly declared it for a host (e.g., `127.0.0.1 ansible_connection=local`). I don't think it will be defined when `--connection=local` is passed, nor if `connection: local` is a playbook parameter. It is maddening.

So, I fear the best option still is something like (untested):
```
when: not sshd_permit_root_login and '127.0.0.1' != ansible_ssh_host | default(inventory_hostname)
```
It's hard to believe there isn't some ansible fact to access that indicates that the connection is local.